packages/datastore/tests/src/datastore.spec.ts

```diff
@@ -198,10 +198,15 @@ describe('@phosphor/datastore', () => {
           expect(change.storeId).to.equal(DATASTORE_ID);
           expect(change.change['test-schema-1']).to.not.be.undefined;
           expect(change.change['test-schema-2']).to.be.undefined;
+          // This code id mostly for checking correcting typing
+          // for change events:
+          const data = change.change['test-schema-1']!['my-record']!;
+          expect(data.visible!.previous).to.equal(false);
+          expect(data.visible!.current).to.equal(true);
```

Comment:
@saulshanabrook This bit! Without the typings for the change/patch structures I get errors like this:

```ts
Property 'visible' does not exist on type:
Change<
  TestSchema | {
    id: "test-schema-2";
    fields: {
      enabled: RegisterField<boolean>;
      count: RegisterField<number>;
      keywords: ListField<string>;
    };
  }
>
```

In other words, the `change.change['test-schema-1']` line isn't able to narrow it to `Change<TestSchema>`. Note that the test here was improved a bit to highlight this issue (the two schemas now have different types).

Reply:
Yeah, and the cleanest way I was able to find were with these guys: https://github.com/saulshanabrook/phosphor/pull/1/files#diff-e8ef547444f515352fab0946e93e81e5R648-R676